### DISCUSSION TITLE
Get the real height of nodes

### DISCUSSION THREAD

**Discussion by Andrii-Moshenskyi** [xyflow/xyflow#3464] [Q&A] [answered]
Hi team! I know that this issue has been discussed several times in one form or another, for example here:
- https://github.com/wbkd/react-flow/issues/991
- https://github.com/wbkd/react-flow/issues/5

But I can’t find `__rf` anywhere at all to take the real sizes of the nodes from there.

Could someone just tell me how to reach the node to get this data?

There is nothing even close to `__rf` in the `nodes` state.
`const [nodes, setNodes, onNodesChange] = useNodesState()`

And I also cannot find this data using `useStore`.

**Comment by moklick** [comment, marked as answer]
After nodes get measured, we add a `height` attribute. That's the real height.

**Comment by Andrii-Moshenskyi** [comment]
Indeed, the height of the nodes is added automatically, I just need to get it correctly after the nodes have already been initialized and ReactFlow has calculated it. Thank you!
